crypto/o_time.c

```diff
@@ -47,6 +47,35 @@ typedef struct _ile3 {          /* Copied from ILEDEF.H for Alpha   */
 # endif   /* alpha || ia64    */
 #endif    /* OPENSSL_SYS_VMS  */
 
+time_t OPENSSL_timegm(struct tm *tm)
+{
+    if (1) {
+#if defined(__USE_MISC)
+        return timegm(tm);
+    } else {
+#endif
+        /* Figures it out without mucking the environment */
+        time_t currtime, utc, local;
+        struct tm *timeinfo;
+        int offset;
+
+        time(&currtime);
```

Comment:
It can't be right to use the current time here.
Maybe, (if an API like that is really absolutely necessary)
use OPENSSL_gmtime_diff and compute the difference between tm and 1970-01-01T00:00:00 ?